### PR TITLE
feat(config): Env var config can now add new items

### PR DESCRIPTION
**Pull request by docmerlin**
This allows for example: adding new headers to httpposts via
environmental variables

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -640,8 +638,28 @@ func (c *Config) applyEnvOverridesToStruct(prefix string, s reflect.Value) error
 					}
 				}
 			} else if f.Kind() == reflect.Map {
-				for _, k := range f.MapKeys() {
-					if err := c.applyEnvOverridesToMap(fmt.Sprintf("%s_%v", key, k), fieldName, f, k, f.MapIndex(k)); err != nil {
+				keys := map[string]reflect.Value{}
```
Comment (lesam) [inline server/config.go:641]: Is this only for `kapacitor.conf` ?

**Review by lesam — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,57 @@ module github.com/influxdata/kapacitor
 
 go 1.17
 
+require (
```
[the remark this answers is not in the document]
Comment (docmerlin) [reply go.mod:5]: `go mod tidy` the new version sorts differently, and maintains sorting.

File: server/config.go (hunk 1 above)
Comment (docmerlin) [reply server/config.go:641]: Yes.

**Comment by timhallinflux** [comment]
@docmerlin does this close: https://github.com/influxdata/kapacitor/issues/2280 ?

**Comment by docmerlin** [comment]
@timhallinflux yes.

**Pull request merged**